Info-Orbs/src/widgets/webDataWidget.cpp

```diff
@@ -34,12 +34,56 @@ void WebDataWidget::draw(bool force) {
             display.setTextDatum(MC_DATUM);
             int yOffset = 110;
 
-            String wrappedLines[MAX_WRAPPED_LINES];
-            String dataValues = data->getData();
-            int lineCount = Utils::getWrappedLines(wrappedLines, dataValues, 10);
-            int height = display.fontHeight() + 10;
-            for (int i = 0; i < lineCount; i++) {
-                display.drawString(wrappedLines[i], 120, yOffset + (height*i), 2);
+            if (data->getElementsCount() > 0) {
+                for (int i = 0; i < data->getElementsCount(); i++) {
+                    WebDataElementModel element = data->getElement(i);
+                    switch (element.getTypeEnum()) {
```

Comment:
If you create a type per item, then this can just be `element->draw()` or something like that,
and the item itself can handle the drawing.